patchwork-registries/src/main/java/com/patchworkmc/mixin/registries/MixinChunkStatus.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.mixin.registries;
+
+import net.minecraftforge.registries.IForgeRegistryEntry;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.minecraft.world.chunk.ChunkStatus;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import com.patchworkmc.impl.registries.ExtendedForgeRegistryEntry;
+
+@Mixin(ChunkStatus.class)
+public class MixinChunkStatus implements ExtendedForgeRegistryEntry<ChunkStatus> {
+	@Unique
+	private Identifier registryName;
+
+	@Override
+	public IForgeRegistryEntry setRegistryName(Identifier name) {
+		this.registryName = name;
+
+		return this;
+	}
+
+	public Identifier getRegistryName() {
+		Identifier current = Registry.CHUNK_STATUS.getId((ChunkStatus) (Object) this);
+		Identifier set = registryName;
```

Comment:
We can really shorten this with `registryName == null ? Registry.CHUNK_STATUS.getDefaultId() : registryName;` tbh

Reply:
It won't have the same effect will it?

Right now the priority would be current -> registryName -> default.
With yours it would be registryName -> default.